src/dawn/CodeGen/Cuda/CudaCodeGen.h

```diff
@@ -35,6 +35,16 @@ class OptimizerContext;
 namespace codegen {
 namespace cuda {
 
+namespace impl_ {
```

Comment:
what is the reasoning behind this namespacing here? Would it not be nicer if this was a private member of the class (as for example here:)
https://github.com/MeteoSwiss-APN/dawn/blob/74fa690bbe58016cc3a6bdf091e9b51ab89e78c7/src/dawn/IIR/StencilMetaInformation.h#L35

Reply:
done